Tavisca.Bootcamp.LanguageBasics.Exercise3/Program.cs

```diff
@@ -28,20 +29,115 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(int[] protein, int[] carbs, int[] fat, string[] dietPlans, int[] expected)
-        {
+        private static void Test(int[] protein, int[] carbs, int[] fat, string[] dietPlans, int[] expected)
+        {
             var result = SelectMeals(protein, carbs, fat, dietPlans).SequenceEqual(expected) ? "PASS" : "FAIL";
             Console.WriteLine($"Proteins = [{string.Join(", ", protein)}]");
             Console.WriteLine($"Carbs = [{string.Join(", ", carbs)}]");
-            Console.WriteLine($"Fats = [{string.Join(", ", fat)}]");
+            Console.WriteLine($"Fats = [{string.Join(", ", fat)}]");
             Console.WriteLine($"Diet plan = [{string.Join(", ", dietPlans)}]");
-            Console.WriteLine(result);
+            Console.WriteLine(result);
+        }
+
+        public static int FindMax(int[] arr, List<int> index)  {
+            int m = arr[index[0]];
+            if(index.Count != 1)  {
+                foreach(int x in index)  {
+                    if(arr[x] > m)  {
+                        m = arr[x];
+                    }
+                }
+            }
+            return m;
+        }
+
+        public static int FindMin(int[] arr, List<int> index)  {
+            int m = arr[index[0]];
+            if(index.Count != 1)  {
+               foreach(int x in index)  {
+                    if(arr[x] < m)  {
+                        m = arr[x];
+                    }
+                }
+            }
+            return m;
         }
 
-        public static int[] SelectMeals(int[] protein, int[] carbs, int[] fat, string[] dietPlans)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public static List<int> FindAllIndex(int[] arr, List<int> index, int elem)  {
+            List<int> l = new List<int>();
+            foreach(int x in index)  {
+                if(arr[x] == elem)  {
+                    l.Add(x);
+                }
+            }
+            return l;
+        }
+
+        public static int[] SelectMeals(int[] protein, int[] carbs, int[] fat, string[] dietPlans)
+        {
+            int[] calorie = new int[protein.Length];
+
+            for(int i=0;i<carbs.Length;i++)  {
+                calorie[i] = ((protein[i] + carbs[i]) * 5) + (fat[i] * 9);
+            }
+
+            int[] meal = new int[dietPlans.Length];
+
+            for(int i=0;i<dietPlans.Length;i++)  {
+
+                List<int> index = new List<int>(protein.Length); 

```

Comment:
Is there an alternative to using `index`?

Reply:
Done